pkg/kernel/stages.go

```diff
@@ -0,0 +1,45 @@
+package kernel
+
+import "sort"
+
+type stages map[int]*stage
+
+func (s stages) hasModules() bool {
+	// no need to iterate in order as we are only checking
+	for _, stage := range s {
+		if len(stage.modules.modules) > 0 {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s stages) countForegroundModules() int32 {
+	count := int32(0)
+
+	// no need to iterate in order as we are only counting
+	for _, stage := range s {
+		for _, m := range stage.modules.modules {
+			if !m.config.background {
+				count++
+			}
+		}
+	}
+
+	return count
+}
+
+func (s stages) getIndices() []int {
+	keys := make([]int, len(s))
```

Comment:
keys := maps.Keys(s)
slices.Sort(keys)

return keys

Reply:
There are also other helpers in maps, slices and funk; but not all of them provide better readability in your cases here

Reply:
I've just moved this code and don't want to change it now.. :)